core/util.bzl

```diff
@@ -47,6 +47,70 @@ def cp(repository_ctx, src, dest = None):
 
     return dest
 
+def cp_many(repository_ctx, srcs):
+    """Copy the given files into the external repository root.
+
+    If possible this is done by invoking tar from an external shell process.
+    This should be more efficient than writing files with repository_ctx.file
+    and preserving their permissions by invoking chmod on every file.
+
+    Args:
+      repository_ctx: The repository context of the current repository rule.
+      srcs: A list of source files to copy.
+
+    Returns:
+      A list of destination paths for the files copied into the repository.
+    """
+    tar_list = []
+    repository_files = []
+
+    # Tar will run with outputBase as its working directory.
+    repository_dir = repository_ctx.execute(["pwd"]).stdout.strip("\n")
+    output_base = repository_dir.removesuffix("external/" + repository_ctx.name)
```

Comment:
Why does `tar` need to run in `outputBase`?

Reply:
I see, makes sense. I guess with GNU tar one could also use `--transform`, but I'm not sure if BSD tar has such an option.